tests/kafkatest/services/verifiable_producer.py

```diff
@@ -167,7 +167,7 @@ def num_not_acked(self):
             return len(self.not_acked_values)
 
     def stop_node(self, node):
-        self.kill_node(node, clean_shutdown=False, allow_fail=False)
```

Comment:
If this is allowed to shutdown gracefully now, should the subsequent `join()` on the worker thread have a timeout? Otherwise it could hang indefinitely if the worker thread doesn't exit properly. I think some of these unlimited timeouts have carried over from some initial test code I wrote originally, but they really should have timeouts.